posts/zzz_DO_NOT_EDIT_introducing.../introducing_sdtm.oak.qmd

```diff
@@ -0,0 +1,254 @@
+      "AE.AETERM"
+    ),
+    paste("VS.VSPOS<br>", "VS.VSLAT"),
+    paste("MH.MHSTDTC<br>", "AE.AEENDTC"),
+    paste(
+      "MH.MHPRESP = 'Y'<br>",
+      "<br>VS.VSTEST = 'Systolic Blood Pressure'<br>",
+      "<br>VS.VSORRESU = 'mmHg'<br>"
+    ),
+    paste(
+      "FA.FASCAT = 'COVID-19 PROBABLE CASE'<br>",
+      "<br>CM.CMTRT = 'FLUIDS'"
+    ),
+    paste(
+      "If If MDPRIOR == 1 then CM.CMSTRTPT = 'BEFORE'.<br>",
+      "<br>VS.VSMETHOD when VSTESTCD = 'TEMP'<br>",
+      "<br>If collected value in raw variable DOS is numeric then CM.CMDOSE<br>",
+      "<br>If collected value in raw variable MOD is different to CMTRT then map to CM.CMMODIFY"
+    )
+  ), stringsAsFactors = TRUE
+)
+knitr::kable(algorithms)
+```
+
+Here is an example of reusing an algorithm across multiple domains,
+variables, and also to a non-standard
+
+![](reusable_algorithms.jpg){width="600"}
+
+# Functions and Parameters
+
+All the aforementioned algorithms are implemented as R functions, each accepting the raw dataset, raw variable, target SDTM dataset, and target SDTM variable as parameters.
+
+```{r}
+library(sdtm.oak)
+library(dplyr)
+
+cm_raw <- tibble::tribble(
+  ~oak_id, ~raw_source, ~patient_number, ~MDRAW, ~DOSU, ~MDPRIOR,
+  1L, "cm_raw", 375L, "BABY ASPIRIN", "mg", 1L,
+  2L, "cm_raw", 375L, "CORTISPORIN", "Gram", 0L,
+  3L, "cm_raw", 376L, "ASPIRIN", NA, 0L
+)
+
+study_ct <- tibble::tribble(
+  ~codelist_code, ~term_code, ~term_value, ~collected_value, ~term_preferred_term, ~term_synonyms,
+  "C71620", "C25613", "%", "%", "Percentage", "Percentage",
+  "C71620", "C28253", "mg", "mg", "Milligram", "Milligram",
+  "C71620", "C48155", "g", "g", "Gram", "Gram"
+)
+
+cm <-
+  # Derive topic variable
+  # SDTM Mapping - Map the collected value to CM. CMTRT
+  assign_no_ct(
+    raw_dat = cm_raw,
+    raw_var = "MDRAW",
+    tgt_var = "CMTRT"
+  ) %>%
+  # Derive qualifier CMDOSU
+  # SDTM Mapping - Map the collected value to CM. CMDOSU
+  assign_ct(
+    raw_dat = cm_raw,
+    raw_var = "DOSU",
+    tgt_var = "CMDOSU",
+    ct_spec = study_ct,
+    ct_clst = "C71620",
+    id_vars = oak_id_vars()
+  ) %>%
+  # Derive qualifier CMSTTPT
+  # SDTM mapping - If MDPRIOR == 1 then CM.CMSTTPT = 'SCREENING'
+  hardcode_no_ct(
+    raw_dat = condition_add(cm_raw, MDPRIOR == "1"),
+    raw_var = "MDPRIOR",
+    tgt_var = "CMSTTPT",
+    tgt_val = "SCREENING",
+    id_vars = oak_id_vars()
+  )
+```
+
+As you can see in this function call, the raw dataset and variable names are passed as arguments. As long as the raw dataset and variable are present in the global environment, the function will execute the algorithm's logic and create the target SDTM variable.
+
+{sdtm.oak} is designed to handle any type of raw input format. It is not tied to any specific data collection standards, making it both EDC-agnostic and data standards-agnostic.
+
+# Why not use {dplyr}?
+
+As you can see from the definition of the algorithms, all of them are a form of mutate statement.
```

Comment:
```suggestion
As you can see from the definition of the algorithms, all of them are a form of {dplyr::mutate()} statement.
```

I think this would be the way for the link package to properly link the function directly. 
Please verify.